lib/spack/llnl/util/tty/__init__.py

```diff
@@ -65,20 +134,37 @@ def process_stacktrace(countback):
     return st_text
 
 
+def get_debug_text():
+    """When debugging is enabled, return a formatted string with milliseconds
+    since epoch"""
+    if _debug:
+        return '[%s] ' % int(round(time.time() * 1000))
```

Comment:
Can we make this readable by humans?  Specifically, how about:

```python
datetime.now().strftime("%Y-%m-%d-%H:%M:%S.%f")
```

Note that `%f` is microseconds, so it's actually finer-grained than what you suggested.  But, since we're this long already, why not.  You can easily read this back with:

```python
datetime.strptime(<string>, "%Y-%m-%d-%H:%M:%S.%f")
```